internal/daysteps/daysteps.go

```diff
@@ -1,20 +1,76 @@
 package daysteps
 
 import (
+	"fmt"
+	"log"
+	"strconv"
+	"strings"
 	"time"
-)
 
-const (
-	// Длина одного шага в метрах
-	stepLength = 0.65
-	// Количество метров в одном километре
-	mInKm = 1000
+	"github.com/Aqouet/go1fl-4-sprint-final/internal/common"
+	"github.com/Aqouet/go1fl-4-sprint-final/internal/spentcalories"
 )
 
+// parsePackage parses a string like "678,0h50m" to number of steps and duration.
+// Returns errors in case of parsing errors or if expected limits of parameters are exceeded.
 func parsePackage(data string) (int, time.Duration, error) {
-	// TODO: реализовать функцию
+
+	dataSl := strings.Split(data, ",")
+	if len(dataSl) != 2 {
+		return 0, time.Duration(0), fmt.Errorf("slice after string %q split = %v with length %d, expected length = 2: %w", data, dataSl, len(dataSl), common.ErrSliceLen)
+	}
+
+	steps, err := strconv.Atoi(dataSl[0])
+	if err != nil {
+		return 0, time.Duration(0), fmt.Errorf("string representing steps = %q: %w", dataSl[0], common.ErrParseInt)
+	}
+
+	if steps <= 0 {
+		return 0, time.Duration(0), fmt.Errorf("steps number = %d,  expected non-negative and non-zero steps number: %w", steps, common.ErrParamLimitExceeded)
+	}
+
+	duration, err := time.ParseDuration(dataSl[1])
+	if err != nil {
+		return 0, time.Duration(0), fmt.Errorf("string representing duration = %q: %w", dataSl[1], common.ErrParseDuration)
```

Comment:
Теряешь сообщение об ошибки
```suggestion
		return 0, 0, fmt.Errorf("conversion error: %w", err)
```